Update dependencies

mikejritter opened the pull request:
Jira: ? 

# Changes

* Update spring dependencies for 5 support
* Replace wix with testcontainers for mysql testing
* Bump easymock and hibernate dependencies
* Set jboss-logging version to avoid dependency conflict

# Notes

I had to do some configuration on my environment to get testcontainers to run properly for me because my docker ran as root only previously. This makes testing a little harder by default and adds an external dependency, but also allows use of an actual mysql db. I don't believe there should be any changes necessary for the gitlab ci to work correctly. @dbernstein let me know what you think.

There were a few small changes I had to make to the tests to get everything to work. Notably, setting the charset and collate for the mill database and removing any data saved after tests were run (I think the embedded db was being recreated on each test before).

There are also a few small api updates from the spring-data updates, I think `findOne` -> `findById` is the most significant.

mikejritter committed:
Update spring to 5

mikejritter committed:
Update easymock version

mikejritter committed:
Spring data api updates

mikejritter committed:
Rollback hibernate version, add testcontainers

mikejritter committed:
Replace wix with testcontainers

mikejritter committed:
Ensure mill uses utf8 encoding

mikejritter committed:
Remove redundant groupId declaration

mikejritter commented:
Something to note that I didn't catch until now:
* The hibernate-validator version specified doesn't exist, and I'm guessing the build passes because it is isn't used anywhere
* The latest version of 5.4.x (which is EOL) for the validator is 5.4.3
* For this project is seems like we might want to drop it from the dependency management, then we can figure out what to do in other projects

dbernstein commented:
I removed the DOCKER_HOST environment variable from my environment and everything built smoothly.

mikejritter committed:
Update spring framework and security versions

mikejritter committed:
Bump mysql-connector version

mikejritter committed:
Bump commons-lang3 to 3.12.0

mikejritter committed:
Bump logback version

mikejritter committed:
Update mysql driver class name

mikejritter committed:
Bump testcontainers version

mikejritter committed:
Remove docker image name from container

I was having problems with the 5.7 mysql container starting cleanly, so I decided to try the default image testcontainers uses. This seems to have fixed my problem with the added benefit of staying in line with the version of testcontainers we're on.

mikejritter committed:
Bump spring versions to latest point release
